### DISCUSSION TITLE
Support for i3status-rust (or rather swaybar)

### DISCUSSION THREAD

**Discussion by mogambro** [YaLTeR/niri#898] [Ideas]
Ignore what I wrote here, I am an idiot :)

**Comment by YaLTeR** [comment]
Hi, Wayland bars should just work, X11 bars won't. I'm not sure if they would work even with Xwayland.

**Reply by mogambro** [reply]
I3status-rust  works on wayland. I used it for years with sway. It just uses this I3 bar protocol.

**Reply by Alanazane** [reply]
[i3bar-river](https://github.com/MaxVerevkin/i3bar-river) might be similar to what you're looking for?  It mentions both i3status-rs and niri support in the readme so it could be worth a try.

**Reply by mogambro** [reply]
@Alanazane thank you so much, this is a late christmas gift. Not sure how I couldn't find this. It's awesome!
